core/src/main/scala/org/apache/spark/sql/catalyst/catalog/TiCatalog.scala

```diff
@@ -144,7 +142,13 @@ class TiCatalog extends TableCatalog with SupportsNamespaces {
     }
 
   override def listNamespaces(): Array[Array[String]] =
-    meta.get.getDatabases.map(dbInfo => Array(dbInfo.getName)).toArray
+    meta.get.getDatabases
+      .filter(db =>
+        if (TiAuthorization.enableAuth) {
```

Comment:
!TiAuthorization.enableAuth || tiAuthorization.visible(db.getName, "")

Reply:
I don't have strong preference over this style. Feel free to ignore this and the following suggestions like so.

Reply:
ACK